tests/functional/requirments.txt

```diff
@@ -0,0 +1,2 @@
+tenacity==9.0.0
+pyyaml==6.0.2
```

Comment:
As general guidance, we try to not pin libs if we don't have a good reason for it like breaking compatibility or another major issue. Of course there are of course downsides of not pinning, but at least we try to be as update as possible.

@aieri can you confirm that?